### PR TITLE
feature(entities): add methods to enable/disable contained comments

### LINKED ISSUE #7516: Previously "ON" comments turned to "OFF" still shows in the river
When a blog owner sets previously "ON" comments to "OFF"  the comments are not shown in the content page, which is okay and correct BUT the comments still show in the river -expected behavior is comments should no longer be shown in the river, as the purpose of the content-owner is now not to show the comments anymore to anyone. The "disappearance" of comment is expected to be in sync.

Using latest 1.9.x Elgg, and just core modules.

### PR DESCRIPTION
**Pull request by mrclay**
Uses these to disable comments on blogs when the comments are turned off.

Fixes #7516 
- [x] needs to turn off access control

### CODE REVIEW

**Comment by hypeJunction** [comment]
Fine by me.

File: engine/classes/ElggEntity.php [hunk 1]
```diff
@@ -559,6 +559,49 @@ public function enableMetadata($name = '') {
 	}
 
 	/**
+	 * Disable all comments contained by this entity
+	 *
+	 * @return void
+	 * @since 2.3
+	 */
+	public function disableComments() {
```
Comment (mrclay) [inline engine/classes/ElggEntity.php]: These methods need to ignore access.

**Comment by hypeJunction** [comment]
Could we make it into a convention for all entity types to use? I don't like this `On`/`Off` deal.

**Comment by mrclay** [comment]
I don't think I want to dig into this during 2.x

**Pull request closed without merging**